core/src/bytecode/value/mod.rs

```diff
@@ -1546,6 +1591,49 @@ impl ValueBlockRc {
         self.try_make_mut().unwrap()
     }
 
+    /// Make a unique (non shared) value block out of `self`, that is a 1-reference counted block
+    /// with the same data. Similar to [Self::make_mut] in spirit but it doesn't require to specify
+    /// the `T` and doesn't return a mutable reference.
+    ///
+    /// This is a no-op if `self` is 1-reference counted. Otherwise, a new (deep) copy is allocated
+    /// and returned.
+    pub fn make_unique(self) -> Self {
+        if self.header().ref_count() == 1 {
+            self
+        } else {
+            self.deep_copy()
+        }
+    }
+
+    /// Creates a copy of the content of this value block. As opposed to [Self::clone], which just
+    /// increments the reference count but encapsulate a pointer to the same block in memory (as
+    /// [std::rc::Rc::clone], this method actually allocates a fresh block with a copy of the
+    /// content and return a value that is 1-reference counted.
+    pub fn deep_copy(&self) -> Self {
```

Comment:
I usually think "deep copy" means that it also descends recursively and deep copies all references (e.g. https://docs.python.org/3/library/copy.html).

Reply:
That's fair, I'll try to come up with a better name.